### ISSUE TITLE
Boolean connection option property no longer works as before 

### ISSUE THREAD

**Issue by languy** [microsoft/azuredatastudio#24078]
- Azure Data Studio Version: 1.45.0
- OS Version: Win11

Steps to Reproduce:
1. Click on new connection button
2. Select Browse to browse your Azure resources
3.  Select a regular Cosmos DB for Mongo API account (not a Cosmos DB vcore mongo)
4. The information in Connection Details will be automatically filled: "Cluster Server" is set as "True", but it should be set as "False".
<img width="371" alt="image" src="https://github.com/Azure/azure-cosmosdb-ads-extension/assets/21954022/866d058a-954e-4078-b058-ded9919dca32">

Root cause:
It appears that [this change](https://github.com/microsoft/azuredatastudio/pull/22512/files) no longer works. It sets the `isServer` as `false` boolean, but the "Connection Details" UI will still display as "True".
If I modify the code to set `isServer` to a string `"false"` or `"true"` and perform checks against a string (e.g. `if (isServer === "true")`) it will work as expected.

For reference:
* the property is defined [here](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/642bb3e4b87f025f1672b35b65104ce595584615/package.json#L300).
* In the extension, it is used [here](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/642bb3e4b87f025f1672b35b65104ce595584615/src/Dashboards/homeDashboard.ts#L35) and [here](https://github.com/Azure/azure-cosmosdb-ads-extension/blob/642bb3e4b87f025f1672b35b65104ce595584615/src/Dashboards/homeDashboard.ts#L44)

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by kenvanhyning** [comment]
Does the connect behavior work properly based on the displayed default value?  I assume the setting is still working if manually changed?  If so, I'd be inclined to move this to next release.  If connect is broken then we'd take a hotfix here.

**Comment by cheenamalhotra** [comment]
Hi @languy 

I tested the commit https://github.com/microsoft/azuredatastudio/commit/e0d5cd18b91b9ceede895cff439d640000c29174 when PR https://github.com/microsoft/azuredatastudio/pull/22512 was merged, the issue exists there too and it seems this use case wasn't validated before. Since the users can always toggle the dropdown to connect, it shouldn't be a release blocker.

I will have a fix out for `main` targeting next release shortly.
